src/Uno.Extensions.Configuration/ConfigurationBuilderExtensions.cs

```diff
@@ -1,43 +1,73 @@
+	/// Adds a JSON configuration source by name to builder. 
+	/// The resultant configuration file name is suffixed with the current hosting environment name.
+	/// </summary>
+	/// <param name="configurationBuilder">The builder instance used to create an <see cref="IConfiguration"/> with keys and values from a set of sources</param>
+	/// <param name="hostingContext">The <see cref="HostBuilderContext"/> which provides information specific to the hosting environment</param>
+	/// <param name="configurationName">A name to identify the added configuration source. Optional</param>
+	/// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
 	public static IConfigurationBuilder AddEnvironmentConfiguration(this IConfigurationBuilder configurationBuilder, HostBuilderContext hostingContext, string? configurationName = null)
 	{
 		var env = hostingContext.HostingEnvironment;
 		var configSection = configurationName is { Length: > 0 } ? $"{configurationName}.{env.EnvironmentName}" : env.EnvironmentName;
 		return configurationBuilder.AddConfigurationFile(hostingContext, string.Format(AppConfiguration.FileNameTemplate, configSection).ToLower());
 	}
 
+	/// <summary>
+	/// Adds a JSON configuration source with the default name to builder.
+	/// </summary>
+	/// <param name="configurationBuilder">The builder instance used to create an <see cref="IConfiguration"/> with keys and values from a set of sources</param>
+	/// <param name="hostingContext">The <see cref="HostBuilderContext"/> which provides information specific to the hosting environment</param>
+	/// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
 	public static IConfigurationBuilder AddAppConfiguration(this IConfigurationBuilder configurationBuilder, HostBuilderContext hostingContext)
 	{
 		return configurationBuilder.AddConfigurationFile(hostingContext, AppConfiguration.FileName);
 	}
 
+	/// <summary>
+	/// Adds a JSON configuration source with the default name suffixed containing the current hosting environment name to builder.
+	/// </summary>
+	/// <param name="configurationBuilder">The builder instance used to create an <see cref="IConfiguration"/> with keys and values from a set of sources</param>
+	/// <param name="hostingContext">The <see cref="HostBuilderContext"/> which provides information specific to the hosting environment</param>
+	/// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
 	public static IConfigurationBuilder AddEnvironmentAppConfiguration(this IConfigurationBuilder configurationBuilder, HostBuilderContext hostingContext)
 	{
 		return configurationBuilder.AddEnvironmentConfiguration(hostingContext, default);
 	}
 
-	public static IConfigurationBuilder AddEmbeddedConfigurationFile<TApplicationRoot>(this IConfigurationBuilder configurationBuilder, string configurationFileName)
+	private static IConfigurationBuilder AddEmbeddedConfigurationFile<TApplicationRoot>(this IConfigurationBuilder configurationBuilder, string configurationFileName)
```

Comment:
This needs to be public as there needs to be a way that you can specify the exact configuration file you want to load from.
AddEmbeddedConfiguration 
- if configurationName is null, will use appsettings.json instead
- if configurationName is not null, will use appsettings.configurationName.json

AddEmbeddedConfigurationFile will use the file called configurationFileName (exactly as it's provided)

Reply:
I've addressed this with the latest commits and annotated this method with the behavior described above